domain/domain_test.go

```diff
@@ -159,27 +159,16 @@ func TestInfo(t *testing.T) {
 	}
 
 	// Test the scene where syncer.Done() gets the information.
-	err = failpoint.Enable("github.com/pingcap/tidb/ddl/util/ErrorMockSessionDone", `return(true)`)
-	if err != nil {
-		t.Fatal(err)
+	if !dom.SchemaValidator.IsStarted() {
+		t.Fatal("syncer is not start")
 	}
-	<-dom.ddl.SchemaSyncer().Done()
-	err = failpoint.Disable("github.com/pingcap/tidb/ddl/util/ErrorMockSessionDone")
+	// Note: We do not check if the syncer is stopped and then restart here,
+	// but the following code ensure the SchemaSyncer restart successfully
+	err = dom.ddl.SchemaSyncer().Close()
 	if err != nil {
 		t.Fatal(err)
 	}
-	time.Sleep(15 * time.Millisecond)
-	syncerStarted := false
-	for i := 0; i < 200; i++ {
-		if dom.SchemaValidator.IsStarted() {
```

Comment:
> Also, because it's hard to detect the state changes of dom.SchemaValidator.IsStarted() from true -> false and false -> true, I remove this check. Even if the check is removed, the code next to syncer.Close() will implicit do the check, because without a running SchemaValidator, these code cannot run and will throw an error.


I don't quite understand why it is difficult to check the value of ʻIsStarted`. Besides. here we want to know the `SchemaValidator` can be restarted automatically after the session is done.